patch-package
--- a/node_modules/dynamoose/lib/Query.js
+++ b/node_modules/dynamoose/lib/Query.js
@@ -87,6 +87,7 @@ Query.prototype.exec = function (next) {
   // the hash and range key on the primary table.  If they don't match then we
   // can look for a secondary index to query.
   if(schema.hashKey.name !== this.query.hashKey.name ||
+    (this.query.rangeKey && !schema.rangeKey) ||
     (this.query.rangeKey && schema.rangeKey && schema.rangeKey.name !== this.query.rangeKey.name)) {
     debug('query is on global secondary index');
     for(indexName in schema.indexes.global) {
